nixos/modules/services/databases/ferretdb.nix

```diff
@@ -0,0 +1,97 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.ferretdb;
+in
+{
+
+  meta.maintainers = with lib.maintainers; [ julienmalka camillemndn ];
+
+  options = {
+    services.ferretdb = {
+      enable = mkEnableOption (mdDoc "FerretDB, a truly Open Source MongoDB alternative.");
+
+      package = mkOption {
+        type = types.package;
+        example = literalExpression "pkgs.ferretdb";
+        default = pkgs.ferretdb;
+        defaultText = "pkgs.ferretdb";
+        description = lib.mdDoc "FerretDB package to use.";
+      };
+
+      user = mkOption {
+        type = types.str;
+        default = "ferretdb";
+        description = lib.mdDoc "User account under which FerretDB runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "ferretdb";
+        description = lib.mdDoc "Group under which FerretDB runs.";
+      };
+
+      settings = lib.mkOption {
+        type =
+          lib.types.submodule { freeformType = with lib.types; attrsOf str; };
+        default = {
+          FERRETDB_POSTGRESQL_URL = "postgres://localhost/ferretdb?host=/run/postgresql";
+        };
+        example = {
+          FERRETDB_LOG_LEVEL = "warn";
+          FERRETDB_MODE = "normal";
+        };
+        description = lib.mdDoc ''
+          Additional configuration for FerretDB, see
+          <https://docs.ferretdb.io/flags/>
+          for supported values.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable
+    {
+
+      systemd.services.ferretdb = {
+        description = "FerretDB";
+        after = [ "network.target" "postgresql.service" ];
+        wantedBy = [ "multi-user.target" ];
+        environment = cfg.settings;
+        serviceConfig = {
+          Type = "simple";
+          StateDirectory = "ferretdb";
+          WorkingDirectory = "/var/lib/ferretdb";
+          ExecStart = "${cfg.package}/bin/ferretdb";
+          Restart = "on-failure";
+          User = cfg.user;
+          Group = cfg.group;
+        };
+      };
+
+      users.users = mkIf (cfg.user == "ferretdb") {
+        ferretdb = {
+          isNormalUser = true;
+          group = cfg.group;
+          home = "/var/lib/ferretdb";
+        };
+      };
+
+      users.groups.ferretdb = mkIf (cfg.group == "ferretdb") { };
+
+      services.postgresql = {
+        enable = true;
+        ensureDatabases = [ "ferretdb" ];
+        ensureUsers = [{
+          name = cfg.user;
+          ensurePermissions."DATABASE ferretdb" = "ALL PRIVILEGES";
+        }];
+      };
+
+      systemd.services.postgresql.environment.LC_ALL = "en_US.UTF-8";
```

Comment:
I'm unsure about this being set on a global level. Can we set it just for one database?

Reply:
I actually removed all references to `postgresql`. `ferretdb` now has several handlers and I believe the user should configure it themselves. I only kept a default on `sqlite` because then it works out of the box.